worlds/rogue_legacy/__init__.py

```diff
@@ -1,176 +1,270 @@
     But that's OK, because no one is perfect, and you don't have to be to succeed.
     """
     game: str = "Rogue Legacy"
-    option_definitions = legacy_options
-    topology_present = False
-    data_version = 3
-    required_client_version = (0, 2, 3)
-    web = LegacyWeb()
+    option_definitions = rl_options
+    topology_present = True
+    data_version = 4
+    required_client_version = (0, 3, 5)
+    web = RLWeb()
 
     item_name_to_id = {name: data.code for name, data in item_table.items()}
-    location_name_to_id = location_table
-
-    def _get_slot_data(self):
-        return {
-            "starting_gender": self.world.starting_gender[self.player],
-            "starting_class": self.world.starting_class[self.player],
-            "new_game_plus": self.world.new_game_plus[self.player],
-            "fairy_chests_per_zone": self.world.fairy_chests_per_zone[self.player],
-            "chests_per_zone": self.world.chests_per_zone[self.player],
-            "universal_fairy_chests": self.world.universal_fairy_chests[self.player],
-            "universal_chests": self.world.universal_chests[self.player],
-            "vendors": self.world.vendors[self.player],
-            "architect_fee": self.world.architect_fee[self.player],
-            "disable_charon": self.world.disable_charon[self.player],
-            "require_purchasing": self.world.require_purchasing[self.player],
-            "gold_gain_multiplier": self.world.gold_gain_multiplier[self.player],
-            "number_of_children": self.world.number_of_children[self.player],
-            "khidr": self.world.khidr[self.player],
-            "alexander": self.world.alexander[self.player],
-            "leon": self.world.leon[self.player],
-            "herodotus": self.world.herodotus[self.player],
-            "allow_default_names": self.world.allow_default_names[self.player],
-            "additional_sir_names": self.world.additional_sir_names[self.player],
-            "additional_lady_names": self.world.additional_lady_names[self.player],
-            "death_link": self.world.death_link[self.player],
-        }
-
-    def _create_items(self, name: str):
-        data = item_table[name]
-        return [self.create_item(name) for _ in range(data.quantity)]
+    location_name_to_id = {name: data.code for name, data in location_table.items()}
+
+    item_pool: List[RLItem] = []
+    prefill_items: List[RLItem] = []
+
+    def setting(self, name: str):
+        return getattr(self.world, name)[self.player]
 
     def fill_slot_data(self) -> dict:
-        slot_data = self._get_slot_data()
-        for option_name in legacy_options:
-            option = getattr(self.world, option_name)[self.player]
+        slot_data = {}
+        for option_name in rl_options:
+            option = self.setting(option_name)
             slot_data[option_name] = option.value
```

Comment:
I would probably write it like this:
```suggestion
        slot_data = {option_name: self.setting(option_name).value for option_name in rl_options}
```

Reply:
Made some tweaks to this to simplifies it more. Thanks!